.github/workflows/python_tests.yml

```diff
@@ -25,7 +25,7 @@ jobs:
       - uses: actions/checkout@v2
       - name: Install dependencies
         run: |
-          apt update && apt install -y socat
+          apt update && apt install -y socat && apt-get install libffi6
```

Comment:
the 38.0.0 wheel is available on https://dl.espressif.com/pypi/cryptography/

you could remove this lib header.